x/docs/internal/keeper/keeper.go

```diff
@@ -40,62 +30,103 @@ func (keeper Keeper) ShareDocument(ctx sdk.Context, document types.Document) {
+	var receivedDocsList types.Documents
 	keeper.cdc.MustUnmarshalBinaryBare(receivedDocs, &receivedDocsList)
 
 	return receivedDocsList
 }
 
 //GetUserSentDocuments returns a list of all documents sent by user
-func (keeper Keeper) GetUserSentDocuments(ctx sdk.Context, user sdk.AccAddress) []types.Document {
+func (keeper Keeper) GetUserSentDocuments(ctx sdk.Context, user sdk.AccAddress) types.Documents {
 	store := ctx.KVStore(keeper.StoreKey)
-	sentDocs := store.Get([]byte(SentDocumentsPrefix + user.String()))
+	sentDocs := store.Get([]byte(types.SentDocumentsPrefix + user.String()))
 
-	var sentDocsList []types.Document
+	var sentDocsList types.Documents
 	keeper.cdc.MustUnmarshalBinaryBare(sentDocs, &sentDocsList)
 
 	return sentDocsList
 }
 
 //TODO Implement these functions when it useful
 
-//Get Document associated with checksum given
-func (keeper Keeper) GetDocument(ctx sdk.Context, checksumValue string) types.Document {
+//Get Document associated with UUID given
+func (keeper Keeper) GetDocument(ctx sdk.Context, uuid string) types.Document {
 	return types.Document{}
 }
 
 // Get all the documents that given sender has shared with given recipient
-func (keeper Keeper) GetSharedDocumentsWithUser(ctx sdk.Context, sender sdk.AccAddress, recipient sdk.AccAddress) []types.Document {
-	return []types.Document{}
+func (keeper Keeper) GetSharedDocumentsWithUser(ctx sdk.Context, sender sdk.AccAddress, recipient sdk.AccAddress) types.Documents {
+	return types.Documents{}
+}
+
+// ShareDocumentReceipt allows to properly store the given receipt
+func (keeper Keeper) ShareDocumentReceipt(ctx sdk.Context, receipt types.DocumentReceipt) {
+	store := ctx.KVStore(keeper.StoreKey)
+
+	//Check if the receipt is already in the store
+	receiptBz := store.Get([]byte(types.DocumentReceiptPrefix + receipt.Uuid + receipt.Recipient.String()))
+	//if it's not, insert it
+	if receiptBz == nil {
+		store.Set([]byte(types.DocumentReceiptPrefix+receipt.Uuid+receipt.Recipient.String()),
+			keeper.cdc.MustMarshalBinaryBare(&receipt))
+	}
+}
+
+// GetUserReceivedReceipts returns the list of all the receipts that the given user has received
+func (keeper Keeper) GetUserReceivedReceipts(ctx sdk.Context, user sdk.AccAddress) types.DocumentReceipts {
+	store := ctx.KVStore(keeper.StoreKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte(types.DocumentReceiptPrefix))
+
+	receivedReceipts := types.DocumentReceipts{}
+	var receipt types.DocumentReceipt
+	for ; iterator.Valid(); iterator.Next() {
+		keeper.cdc.MustUnmarshalBinaryBare(iterator.Value(), &receipt)
+		if receipt.Recipient.Equals(user) {
+			receivedReceipts = receivedReceipts.AppendReceiptIfMissing(receipt)
+		}
+	}
+	return receivedReceipts
+}
+
+// GetReceiptByDocumentUuid returns the receipts that the given recipient has received for the document having the
+// given uuid
+func (keeper Keeper) GetReceiptByDocumentUuid(ctx sdk.Context, recipient sdk.AccAddress, uuid string) types.DocumentReceipt {
```

Comment:
I think this method should have a clearer name, such as `GetUserReceivedReceiptsForDocument`. 

Also, why does this method return a single `types.DocumentReceipts` and not a list of them?
What would happen if a user sends the same document having the same UUID to different people and want to know the list of receipts that have been sent back to him? 

E.g. 

1. Alice sends the Document with UUID `6e7a3509-3f10-43a9-9367-3d5d0c7f28bb` to Bob, Carl and Denver. 
2. Bob and Denver send back the receipt to Alice

Shouldn't Alice be able to get the whole list of received receipts for that document? 

We should add this test case into the test base